src/EnergyPlus/HVACFourPipeBeam.cc

```diff
@@ -0,0 +1,1302 @@
+// C++ Headers
+#include <cmath>
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array.functions.hh>
+#include <ObjexxFCL/Fmath.hh>
+
+// EnergyPlus Headers
+#include <AirTerminalUnit.hh>
+#include <HVACFourPipeBeam.hh>
+#include <BranchNodeConnections.hh>
+#include <DataContaminantBalance.hh>
+#include <DataDefineEquip.hh>
+#include <DataEnvironment.hh>
+#include <DataHVACGlobals.hh>
+#include <DataIPShortCuts.hh>
+#include <DataLoopNode.hh>
+#include <DataPlant.hh>
+#include <DataPrecisionGlobals.hh>
+#include <DataSizing.hh>
+#include <DataZoneEnergyDemands.hh>
+#include <DataZoneEquipment.hh>
+#include <FluidProperties.hh>
+#include <General.hh>
+#include <GeneralRoutines.hh>
+#include <InputProcessor.hh>
+#include <NodeInputManager.hh>
+#include <OutputProcessor.hh>
+#include <PlantUtilities.hh>
+#include <Psychrometrics.hh>
+#include <ReportSizingManager.hh>
+#include <ScheduleManager.hh>
+#include <UtilityRoutines.hh>
+#include <CurveManager.hh>
+
+
+namespace EnergyPlus {
+
+namespace FourPipeBeam {
+
+	Array1D< std::shared_ptr< HVACFourPipeBeam > > FourPipeBeams; // dimension to number of machines
```

Comment:
@DeadParrot The main reason I chose to use shared_ptr in the plant restructuring branch I have is that a single component can live on multiple plant loops, but each loop uses the same entry point into the calculation, so it was a natural fit.  There was a positive response from many folks about using this, but I think we'd all be happy to have a little more discussion before the pattern becomes mainstream.

One comment on this though.  I think we have a little bit of a chicken/egg with regard to trying out these concepts and figuring out what is the best design.  I put out the shared_ptr approach in my branch, and it works great.  I like letting @EnergyArchmage try it here because he will be vocal on whether it seems completely against the grain.  If we find it doesn't work in places, or there is a noticeable performance hit that outweighs the benefit of using them, then we aren't going to be very far down the road in terms of pulling back.  And at that point we would have more data points to determine what would be _better_.

All that said, let's have a chat maybe next week?  Or would it be better suited for immediately after the release?